### ISSUE TITLE
Enhancement: [consistent-type-imports] allow "typeof import()" types with "disallowTypeAnnotations:true"

### ISSUE THREAD

**Issue by danielrentz** [typescript-eslint/typescript-eslint#6474]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#supported-rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/consistent-type-imports

### Description

I need to use the type of `import` expressions (see example), and still want to disallow all other kinds of type imports that are banned with the "disallowTypeAnnotations" option.

Is this possible somehow else?

### Fail

```typescript
// everything that is documented in the rule
type T = import('Foo').Foo;
const x: import('Bar') = 1;
```

### Pass

```typescript
const module = await import(`${path}.json`) as typeof import("*.json"); // { default: string }
return module.default;
```

**Comment by bradzacher** [comment]
You don't *need* to use an inline typeof import - you can do this with an import declarations:
```ts
import type * as JSONModule from '*.json';

async function main(name: string) {
  const foo = (await import(`${name}.json`)) as typeof JSONModule;
}
```

**Comment by danielrentz** [comment]
Yes, that's the solution, thanks!
